sszgen/hash.go

```diff
@@ -95,78 +95,112 @@ func (v *Value) hashRoots(isList bool, elem Type) string {
 	})
 }
 
-func (v *Value) hashTreeRoot() string {
+// takes a "name" param so that the variable name can be replaced with a local name
+// ie within a for loop for a list, the we want to refer to "elem" w/o a receiver variable
+// when not specified, name will be set to "::." + v.name. In the final templating pass,
+// the output formatter replaces all instances of "::" with the receiver variable for the container.
+func (v *Value) hashTreeRoot(name string, appendBytes bool) string {
```

Comment:
Can we add a little more color to this appendBytes option? It's clear that it's used as a control variable to do AppendBytes instead of PutBytes, but would be nice to explain why we do it, or when we do it

Reply:
will do!